bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/extensions/ExtensionEvent.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.rest.core.extensions;
+
+import org.eclipse.smarthome.core.events.AbstractEvent;
+
+/**
+ * This is an {@link Event} that is sent on extension operations, such as installing and uninstalling.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ */
+public class ExtensionEvent extends AbstractEvent {
+
+    /**
+     * The extension event type.
+     */
+    public final static String TYPE = ExtensionEvent.class.getSimpleName();
+
+    private String msg;
+    private String id;
+
+    /**
+     * Constructs a new extension event object.
+     *
+     * @param topic the topic
+     * @param payload the payload
+     * @param id the id of the extension
+     * @param msg the message text
+     */
+    public ExtensionEvent(String topic, String payload, String id, String msg) {
+        super(topic, payload, null);
+        this.id = id;
+        this.msg = msg;
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    @Override
+    public String toString() {
+        return id + ": " + msg;
```

Comment:
Instead of using the message, evaluate the topic to generate a good string message